### ISSUE TITLE
valid_doc false negative

### ISSUE THREAD

**Issue by agordeev** [realm/SwiftLint#758]
The function below returns warning: `Valid Docs Violation: Documented declarations should be valid. (valid_docs)`

```
/**
 Sets alpha to all subviews of a specific `UIView`, except of given subview.

 - parameter alpha: New alpha value for subviews
 - parameter view: UIView the action should be applied to
 - parameter exceptView: The view, that should not change it's alpha
 - parameter duration: Duration of the animation in seconds
 */
class func setAlpha(alpha alpha: CGFloat, view: UIView, exceptSubview: UIView, duration: NSTimeInterval) {
}
```

Is there any way to check what exactly causes this warning?

**Comment by Yaro812** [comment]
check parameter name for exceptView  (it is exceptSubview in your method)

**Comment by andrew8712** [comment]
@Yaro812 Thanks a lot, my bad!

However, this one shows the warning as well:

    /**
     Returns messages for given conversation.

     - parameter conversation: A conversation.
     - parameter limit: Defines max count of returned messages.
     - parameter success: Success closure.
     - parameter failure: Failure closure.
    */
    func obtainPreviousMessagesForConversation(conversation: Conversation, limit: Int, success: SuccessWithEntitiesCompletionHandler, failure: FailureCompletionHandler) {
    }

**Comment by freak4pc** [comment]
Just tested it and not getting a documentation error on it. 

14:26:39 freak4pc » swiftlint                                                                                                                                      

Linting Swift files in current working directory
Linting 't.swift' (1/1)
/Users/freak4pc/Work/Tests/lint/t.swift:9: warning: Line Length Violation: Line should be 100 characters or less: currently 166 characters (line_length)
Done linting! Found 1 violation, 0 serious in 1 file.

**Comment by Yaro812** [comment]
You are using the completion handler. There might be a slight bug in Swiftlint that shows documentation warning when closure that returns no value is written like (param: Int) -> (). You can rewrite it to be (param: Int) -> Void

**Comment by andrew8712** [comment]
@freak4pc Interesting. Can you please try this one:


    /**
     Returns messages for given conversation.

     - parameter conversation: A conversation.
     - parameter limit: Defines max count of returned messages.
     - parameter success: Success closure.
     - parameter failure: Failure closure.
    */
    func obtainPreviousMessagesForConversation(conversation: String, limit: Int, success: (() -> Void)?, failure: (() -> Void)?) {
    }

I'm getting a warning with this.

By the way, my completion handlers looks like this:

    typealias SuccessWithEntitiesCompletionHandler = ((entities: [T]) -> Void)?
    typealias FailureCompletionHandler = ((allowRetry: Bool, errorMessage: String) -> Void)?

If the above piece compiles fine for you, may be I should update my SwiftLint? The current version is 0.10.0.

**Comment by freak4pc** [comment]
I'm a bit busy atm will check later but you should definitely update. Current is 0.11.1 afaik.

**Comment by andrew8712** [comment]
@freak4pc Thanks. `brew` says `swiftlint 0.10.0 already installed`.

**Comment by freak4pc** [comment]
@andrew8712  Did you try `brew update && brew upgrade swiftlint` ?

**Comment by andrew8712** [comment]
@freak4pc No, sorry. Updated fine now, however, getting the same warning with swiftlint 0.11.1

**Comment by andrew8712** [comment]
@freak4pc I found the problem. My func's description starts with `Returns` word, which caused `ValidDocsRule.commentReturns(_:)` function returns `true`. However, the func doesn't return any value, that's why `SwiftLint` is showing a warning: `superfluousReturnDocumentation` returns `true`.

Since my func has completion handlers, I believe the documentation may contain "Returns" keyword. So this condition is not the best fit:

        return comment.lowercaseString.containsString("- returns:") ||
        comment.rangeOfString("Returns")?.startIndex == comment.startIndex

Any ideas how to handle a case with completion handlers?

**Comment by jpsim** [comment]
If you want to bypass SwiftLint's semantics for determining whether or not a function returns, disable the rule.

**Comment by freak4pc** [comment]
You description starts with a `- returns:` ? as an entire phrase ? that seems odd. 
Could you share more details on how that looks in your code?

**Comment by jpsim** [comment]
A description starting with `Returns` is treated the same was as if `- returns: ` had been included.

I don't think blocks/closures that are invoked with parameters count as "returning" in the same sense, and therefore shouldn't use either approach.

**Comment by marcelofabri** [comment]
Closing this since we removed this rule.

**Comment by freak4pc** [comment]
@marcelofabri Mind linking to discussion on that?
Was the valid docs rule removed altogether ?

**Comment by marcelofabri** [comment]
This rule wasn't working at all since Swift 2.3. At some point we disabled it on Swift 2.3+ and on the next release we'll drop support for Swift 2, so the rule was removed.

**Comment by freak4pc** [comment]
Gotya. Thanks for the clarification 👍 
Should we investigate rewriting that rule or the consensus is there's no use for it ATM ?

**Comment by marcelofabri** [comment]
The issue was that SourceKit stopped providing the information we need to make it work. Check #728 for more info.
